gobblin-api/src/main/java/org/apache/gobblin/dataset/PartitionDescriptor.java

```diff
@@ -37,6 +37,13 @@ public PartitionDescriptor copy() {
     return new PartitionDescriptor(getName(), dataset);
   }
 
+  /**
+   * Create a copy of partition descriptor under a new dataset
+   */
+  public PartitionDescriptor copy(DatasetDescriptor dataset) {
```

Comment:
nitpick: `copy()` is a misleading name since we are actually changing the dataset - I think I'd actually just delete this method altogether and just use `new PartitionDescriptor(getName(), dataset)` when you are trying to copy; makes it very obvious

Reply:
What I want to do is to copy everything except the dataset. And subclasses may have more fields. Probably name it`move(DatasetDescriptor dataset)`?

Reply:
how about `cloneWithNewDataset`? or `copyWithNewDataset`?